src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -329,29 +380,72 @@ private boolean intersectionsAreExplicit(final GeometricSurface waterbody,
+     * It's estimated that if the outer member only has the {@link ISOCountryTag} and
+     * {@link SyntheticNearestNeighborCountryCodeTag}, it was created due to atlas slicing and would
+     * therefore not be a geometrically accurate member of the relation. 2. The intersecting feature
+     * is entirely floating in an outer member of the relation, and not intersecting/floating in any
+     * of the inner members. The inner members are typically islands so we avoid flagging in the
+     * latter scenario. The outer member must still have only the 2 tags mentioned above
+     *
+     * @param waterbody
+     *            the GeometricSurface representation of the waterbody
+     * @param object
+     *            the AtlasEntity representation of the waterbody
+     * @param intersectingFeature
+     *            the intersecting building/road/line
+     * @return true if either of the 2 above conditions are satisfied, false otherwise
+     */
+    private boolean intersectsValidPartOfWaterbody(final GeometricSurface waterbody,
+            final AtlasObject object, final PolyLine intersectingFeature)
+    {
+        // The waterbody is not an area, or...
+        return !(waterbody instanceof MultiPolygon)
+                // Map all outer multipolygon members to a tuple of its AtlasEntity and the entity's
+                // geometric Polygon, and check for appropriate intersection/containment/tagging
+                || ((Relation) object).members().stream()
+                        .flatMap(member -> member.getEntity() instanceof Relation
+                                ? ((Relation) member.getEntity()).members().stream()
+                                : Stream.of(member))
+                        .filter(member -> member.getRole().equals("outer"))
+                        .map(member -> new Tuple<>(member.getEntity(),
+                                new Polygon(member.getEntity() instanceof Area
+                                        ? (Area) member.getEntity()
+                                        : (LineItem) member.getEntity())))
+                        .anyMatch(member -> (intersectingFeature.intersects(member.getSecond())
+                                || (intersectingFeature.within(member.getSecond())
+                                        && ((MultiPolygon) waterbody).inners().stream()
+                                                .noneMatch(innerMember -> intersectingFeature
+                                                        .intersects(innerMember)
+                                                        || intersectingFeature
+                                                                .within(innerMember))))
+                                && (member.getFirst().getTags().keySet().stream().anyMatch(
+                                        key -> !(key.equals(ISOCountryTag.KEY) || key.equals(
+                                                SyntheticNearestNeighborCountryCodeTag.KEY)))));
```

Comment:
I think the use of SyntheticNearestNeighborCountryCodeTag in country slicing has been removed by https://github.com/osmlab/atlas/pull/627.

Reply:
Good catch. Taking that out so we now only check if the feature has a tag other than the IsoCountryTag

Reply:
According to https://github.com/osmlab/atlas/blob/f9ebfa344fb5891e3a21a2e790ea3f153230d4ec/src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java#L442, synthetically added relation members at borders would only have the IsoCountryTag tag. So checking for any tag other than that is looking like a good/straightforward estimation for whether or not the feature was synthetically created, especially considering that features not synthetically created would have OSM tags as well.

Reply:
nice! thanks for looking into it

Reply:
So to clarify.. if the feature _only_ contains a iso country code, we can assume it was synthetically added.

Reply:
as far as multipolygon relation members go, that's the idea.